File: shared/src/androidMain/kotlin/com/jw/common/common/kotlinmultilib/preferences/CommonPreference.java
package com.jw.common.common.kotlinmultilib.preferences;

import android.content.Context;
import android.content.SharedPreferences;

import org.jetbrains.annotations.Nullable;
import com.jw.media.lib.utils.provider.ContextProvider;
import java.util.Set;

/**
 * Created by Joyce.wang on 2024/9/11 16:45
 *
 * @Description TODO
 */
public class CommonPreference {
    private static final String TAG = CommonPreference.class.getSimpleName();
    private final static String APP_PREFERENCE_NAME = "jw_media";
    private SharedPreferences mSharedPreferences;

    public static CommonPreference getInstance() {
        return SingletonInnerClass.INSTANCE;
    }

    static class SingletonInnerClass {
        final static CommonPreference INSTANCE = new CommonPreference();
    }
    public CommonPreference() {
        this.mSharedPreferences = ContextProvider.Companion.getContext().getSharedPreferences(APP_PREFERENCE_NAME, Context.MODE_PRIVATE);
    }

    public void putStringSet(String key, @Nullable Set<String> values) {
        mSharedPreferences.edit().putStringSet(key, values).apply();
    }

    public void putString(String key, @Nullable String value) {
        mSharedPreferences.edit().putString(key, value).apply();
    }

    public void putInt(String key, int value) {
        mSharedPreferences.edit().putInt(key, value).apply();
    }

    public void putLong(String key, long value) {
        mSharedPreferences.edit().putLong(key, value).apply();
    }

    public void putFloat(String key, float value) {
        mSharedPreferences.edit().putFloat(key, value).apply();
    }

    public void putBoolean(String key, boolean value) {
        mSharedPreferences.edit().putBoolean(key, value).apply();
    }

    public String getString(String key) {
        return getString(key, "");
    }

    public String getString(String key, @Nullable String defValue) {
        return mSharedPreferences.getString(key, defValue);
    }

    public Set<String> getStringSet(String key, @Nullable Set<String> defValues) {
        return mSharedPreferences.getStringSet(key, defValues);
    }

    public int getInt(String key, int defValue) {
        return mSharedPreferences.getInt(key, defValue);
    }

    public long getLong(String key, long defValue) {
        return mSharedPreferences.getLong(key, defValue);
    }

    public float getFloat(String key, float defValue) {
        return mSharedPreferences.getFloat(key, defValue);
    }

    public boolean getBoolean(String key, boolean defValue) {
        return mSharedPreferences.getBoolean(key, defValue);
    }

    public void remove(String key) {
        mSharedPreferences.edit().remove(key);
    }
}

File: lib_utils/src/main/java/com/jw/media/lib/utils/preferences/CommonPreference.kt
package com.jw.media.lib.utils.preferences

import android.annotation.SuppressLint
import android.content.Context
import android.content.SharedPreferences
import com.jw.media.lib.utils.provider.ContextProvider

/**
 *Created by Joyce.wang on 2024/9/24 15:10
 *@Description TODO
 */
object CommonPreference {
    private val TAG: String = CommonPreference::class.java.simpleName
    private val APP_PREFERENCE_NAME: String = "jw_media"

    private lateinit var mSharedPreferences: SharedPreferences

    init {
        mSharedPreferences = ContextProvider.getContext().getSharedPreferences(APP_PREFERENCE_NAME, Context.MODE_PRIVATE)
    }

    fun putStringSet(key: String, values: Set<String?>?) {
        mSharedPreferences.edit().putStringSet(key, values).apply()
    }

    fun putString(key: String, value: String?) {
        mSharedPreferences.edit().putString(key, value).apply()
    }

    fun putInt(key: String, value: Int) {
        mSharedPreferences.edit().putInt(key, value).apply()
    }

    fun putLong(key: String, value: Long) {
        mSharedPreferences.edit().putLong(key, value).apply()
    }

    fun putFloat(key: String, value: Float) {
        mSharedPreferences.edit().putFloat(key, value).apply()
    }

    fun putBoolean(key: String, value: Boolean) {
        mSharedPreferences.edit().putBoolean(key, value).apply()
    }

    fun getString(key: String): String? {
        return getString(key, "")
    }

    fun getString(key: String, defValue: String?): String? {
        return mSharedPreferences.getString(key, defValue)
    }

    fun getStringSet(key: String, defValues: Set<String?>?): Set<String>? {
        return mSharedPreferences.getStringSet(key, defValues)
    }

    fun getInt(key: String, defValue: Int): Int {
        return mSharedPreferences.getInt(key, defValue)
    }

    fun getLong(key: String, defValue: Long): Long {
        return mSharedPreferences.getLong(key, defValue)
    }

    fun getFloat(key: String, defValue: Float): Float {
        return mSharedPreferences.getFloat(key, defValue)
    }

    fun getBoolean(key: String, defValue: Boolean): Boolean {
        return mSharedPreferences.getBoolean(key, defValue)
    }

    @SuppressLint("CommitPrefEdits")
    fun remove(key: String) {
        mSharedPreferences.edit().remove(key)
    }
}